### PR TITLE
Add stories to generate tokens from connection string

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What

Calling canvas:

image
image

Chat canvas:

image
image

# Why
A step towards improving the stories that currently seem to imply that connection string should be used by the UI library directly.

# How Tested
Local storybook

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by prprabhu-ms** [comment]
The Storybook URL is amazing!

https://60c7ae6891f0e90039d7cd54-eutozoojlu.chromatic.com/?path=/docs/identitymanagement--call-backend is the exact link.

**Comment by JamesBurnside** [comment]
> The Storybook URL is amazing!
> 
> [60c7ae6891f0e90039d7cd54-eutozoojlu.chromatic.com/?path=/docs/identitymanagement--call-backend](https://60c7ae6891f0e90039d7cd54-eutozoojlu.chromatic.com/?path=/docs/identitymanagement--call-backend) is the exact link.

Yesssssssssss!

Two nits:
* Clicking "Chat Backend" scrolls me to the top of the doc (or starts the doc at the top) not at the chat section. Should we just make this story "Generate token utility"
* Anyway we can make the json prettified? (mine is flat with no indentation - but font and color would be nice if storybook supports it

File: packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+};
+
+const CopyableResponse = (props: { response: unknown | undefined }): JSX.Element => {
+  const { response } = props;
+  const theme = useTheme();
+
+  if (response === undefined) {
+    return (
+      <Text>
+        Tip: Connection string can be{' '}
```
Comment (JamesBurnside) [inline packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx:267]: personal preference nit: I'd remove the 'Tip' and just have the Connection string can b...
```suggestion
        Connection string can be{' '}
```

File: packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        <Link href="https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/create-communication-resource?tabs=windows&pivots=platform-azp#access-your-connection-strings-and-service-endpoints">
+          found from the azure portal.
```
Comment (JamesBurnside) [inline packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx:269]: remove period here (there's a second one below)

**Comment by JamesBurnside** [comment]
Would be great to see this linked somehow in the composite quickstarts

File: packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx (hunk 1 above)
Comment (prprabhu-ms) [reply packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx:267]: Done.

**Review by JamesBurnside — APPROVED** [review]

**Comment by prprabhu-ms** [comment]
> > The Storybook URL is amazing!
> > [60c7ae6891f0e90039d7cd54-eutozoojlu.chromatic.com/?path=/docs/identitymanagement--call-backend](https://60c7ae6891f0e90039d7cd54-eutozoojlu.chromatic.com/?path=/docs/identitymanagement--call-backend) is the exact link.
> 
> Yesssssssssss!
> 
> Two nits:
> 
> * Clicking "Chat Backend" scrolls me to the top of the doc (or starts the doc at the top) not at the chat section.

I don't think there's a way to correlate a doc section with a canvas in storybook.

> * Anyway we can make the json prettified? (mine is flat with no indentation - but font and color would be nice if storybook supports it

I think I've spent way too much time on this for now (got a bit carried away with this yesterday...)
Note that the form etc inside the Canvas is just Fluent UI, not storybook. Our of the box, Fluent UI doesn't have any way to render code blocks. Will keep this in mind next time I want to hack around :D

**Comment by prprabhu-ms** [comment]
> Would be great to see this linked somehow in the composite quickstarts

Yep. That's the plan, though it needs more modifications to the composite stories themselves to stop expecting connection string wherever possible. I'm working on just adding these stories in first as a step, because it's already useful for the "external call" and "external chat" canvases.

**Comment by ddematheu2** [comment]
From an organizational perspective, trying to think where is the best place to hang this. Thinking if it might make sense to have under composite header or as its own section.. Thoughts?

**Comment by prprabhu-ms** [comment]
> From an organizational perspective, trying to think where is the best place to hang this. Thinking if it might make sense to have under composite header or as its own section.. Thoughts?

tl;dr -> don't know :(

First off, the current place isn't intentional. I just added it _somewhere_.
I don't think the Composites section is a good fit -- you need the exact same identity management if you're using the stateful clients with UI components.

It's own section seems heavy-handed, what would we call it? Would it be OK to move it under Examples?

**Review by alcail — APPROVED** [review]

File: packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+const CopyableResponse = (props: { response: unknown | undefined }): JSX.Element => {
+  const { response } = props;
+  const theme = useTheme();
+
+  if (response === undefined) {
+    return (
+      <Text>
+        Connection string can be{' '}
+        <Link href="https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/create-communication-resource?tabs=windows&pivots=platform-azp#access-your-connection-strings-and-service-endpoints">
+          found from the azure portal
```
Comment (anjulgarg) [inline packages/storybook/stories/IdentityManagement/IdentityManagement.stories.tsx:269]: generated on Azure Portal.

File: packages/storybook/stories/IdentityManagement/IdentityManagement.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import * as stories from './IdentityManagement.stories';
+
+import CallBackendText from '!!raw-loader!./snippets/CallBackend.snippet.ts';
+import ChatBackendText from '!!raw-loader!./snippets/ChatBackend.snippet.ts';
+
+<Meta id="identitymanagement" title="Identity Managemment" />
+
+# Identity Management
+
+It is a primary tenet of Azure Communication Services that customers bring their own user identities. Customers
```
Comment (anjulgarg) [inline packages/storybook/stories/IdentityManagement/IdentityManagement.stories.mdx:11]: I suggest breaking it down into steps. For example,
- Customers deploy a secure services for generating tokens
- Services  create tokens using Azure... identity services.
- These tokens are supplied to applications....

etc etc...

Comment (anjulgarg) [reply packages/storybook/stories/IdentityManagement/IdentityManagement.stories.mdx:11]: Can you link to the source of this information? `customers bring their own user identities` sounds misleading.

Comment (prprabhu-ms) [reply packages/storybook/stories/IdentityManagement/IdentityManagement.stories.mdx]: Linked to the source in the docs, and reworded to be similar to the starting paragraph there.

Comment (prprabhu-ms) [reply packages/storybook/stories/IdentityManagement/IdentityManagement.stories.mdx]: Also moved to bulleted list.

**Comment by prprabhu-ms** [comment]
Happy to move this around in the contents, but landing this PR for now.
